From a3667c0b30e1938c79a5ad43459d920ab662c912 Mon Sep 17 00:00:00 2001
From: David Kershner <dkershner@habana.ai>
Date: Thu, 12 Jun 2025 17:49:23 +0300
Subject: [PATCH] [SW-226950] oobi: correct use of sizeof in memcpy

When doing memcpy to pldm_sensor_data and pldm_effecter_data, use the
destination size instead of the source size.

Change-Id: Ibe4b9159453145aee7af53b65270b51e5b5455c2
Signed-off-by: David Kershner <dkershner@habana.ai>
---

diff --git a/oobi.c b/oobi.c
index 7963ff9..3bffe6d 100644
--- a/oobi.c
+++ b/oobi.c
@@ -407,7 +407,7 @@
 {
 	struct oobi *info = _get();
 
-	memcpy(&info->v3.pldm_effecter_data, data, sizeof(*data));
+	memcpy(&info->v3.pldm_effecter_data, data, sizeof(info->v3.pldm_effecter_data));
 }
 
 void hl_oobi_pldm_effecter_clear_data(void)
@@ -428,7 +428,7 @@
 {
 	struct oobi *info = _get();
 
-	memcpy(&info->v3.pldm_sensor_data, data, sizeof(*data));
+	memcpy(&info->v3.pldm_sensor_data, data, sizeof(info->v3.pldm_sensor_data));
 }
 
 void hl_oobi_pldm_sensor_clear_data(void)
